### PR TITLE
Improve error message logging on exit

### PR DESCRIPTION
**Pull request by tsimonq2**
This should fix [bug 1574190](https://bugs.launchpad.net/snapcraft/+bug/1574190) due to a suggestion by @elopio. This reverts [a one-line change](https://github.com/ubuntu-core/snapcraft/commit/3e3927a534aece63c992a34818e61f4a596802b5#diff-07ab19acf782b5f8bd05c71b8d04f963R101) made with the commit for `cleanbuild`.

### CODE REVIEW

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by tsimonq2** [comment]
Needs more work, I'll hack on it more

**Comment by tsimonq2** [comment]
Hmm, the unit tests are failing, but I'm not entirely sure that it's because of tweaks I'm making. @elopio, thoughts on why this is failing?

File: snapcraft/main.py [hunk 1]
```diff
@@ -209,9 +208,7 @@ def main(argv=None):
     except Exception as e:
         if args['--debug']:
             raise
-
-        logger.error(textwrap.fill(str(e)))
-        sys.exit(1)
+        sys.exit(e)
```
Comment (elopio) [inline snapcraft/main.py]: This is a bad merge. I changed these lines after you made your branch.
The correct merged statements would be:
logger.error(str(e))
sys.exit(1)

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by snappy-m-o** [comment]
Can one of the admins verify this patch?

**Comment by tsimonq2** [comment]
Well, this looks like the tests have passed and the bug seems to be fixed.

@elopio can you confirm that this is ready?

**Comment by kyrofa** [comment]
Hey @tsimonq2  this turns into a changelog entry. Would you mind selecting a more informative PR title?

**Comment by kyrofa** [comment]
ok to test

**Comment by tsimonq2** [comment]
@kyrofa How does that look?

**Comment by sergiusens** [comment]
@tsimonq2 I hope you don't mind my change

**Comment by tsimonq2** [comment]
Totally fine @sergiusens

**Comment by sergiusens** [comment]
:+1:

**Comment by elopio** [comment]
:+1: Thanks!

**Pull request merged**